### ISSUE TITLE
`weighted_metrics` not showing up next to training progress bar

### ISSUE THREAD

**Issue by abheesht17** [keras-team/keras-core#454]
Tried porting this example (https://keras.io/examples/generative/gpt2_text_generation_with_kerasnlp/) to Keras Core. `accuracy` (passed as a `weighted_metric`) doesn't show up in the training bar (across backends).

Reproduced it here (with JAX): https://colab.research.google.com/drive/1XByfu9-vaFLnXdUjW6tLypMYy36C8LVx?usp=sharing
This is with legacy Keras: https://colab.research.google.com/drive/1fucRLYK_7ZDMRN5GURMgRIe80JknaFhj?usp=sharing

**Comment by fchollet** [comment]
Thanks for the report. Can you repro with a regular Keras Core model (not KerasNLP)?

**Comment by abheesht17** [comment]
> Thanks for the report. Can you repro with a regular Keras Core model (not KerasNLP)?

Here you go: https://colab.research.google.com/drive/1NJsyb_IzR1Lnc2YD1U4AAV-mcutx6DM1?usp=sharing!
